src/Plugin/RemoteCheck/RemoteRegister.php

```diff
@@ -0,0 +1,139 @@
+    $this->serializer = $serializer;
+  }
+
+  /**
+   * @param \Drupal\contentpool_client\Plugin\RemoteCheck\ContainerInterface $container
+   * @param array $configuration
+   * @param $plugin_id
+   * @param $plugin_definition
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('serializer')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute(RemoteInterface $remote) {
+    $url = (string) $remote->uri();
+
+    if(!$remote->getThirdPartySetting('contentpool_client', 'autoregister', 0)) {
+      $this->result = TRUE;
+      $this->message = t('Autoregister inactive.');
+      return;
+    }
+
+    // As the remote targets the relaxed endpoint we have to parse the url
+    // to get the base host.
+    $url_parts = parse_url($url);
+    $credentials = '';
+    if (isset($url_parts['user']) && isset($url_parts['pass'])) {
+      $credentials = $url_parts['user'] . ':' . $url_parts['pass'] . '@';
+    }
+
+    $base_url = $url_parts['scheme'] . '://' . $credentials . $url_parts['host'];
+
+    /** @var \GuzzleHttp\Client $client */
+    $client = \Drupal::httpClient();
+
+    try {
+      $response = $client->post($base_url . '/_remote-registration?_format=json', $this->generateRegistrationPayload());
+
+      if ($response->getStatusCode() === 200) {
+        $this->result = TRUE;
+        $this->message = t('Registration on remote is valid.');
+      }
+      else {
+        $this->message = t('Remote returns status code @status.', ['@status' => $response->getStatusCode()]);
+      }
+    }
+    catch (\Exception $e) {
+      $this->message = $e->getMessage();
+      watchdog_exception('relaxed', $e);
+    }
+  }
+
+  /**
+   * Collects the data necessary for registration on remote.
+   */
+  private function generateRegistrationPayload() {
+    // Basic Site information.
+    $config = \Drupal::config('system.site');
+    $site_name = $config->get('name');
+    $site_uuid = $config->get('uuid');
+    $site_host = \Drupal::request()->getSchemeAndHttpHost();
+
+    $body = [
+      'site_name' => $site_name,
+      'site_domain' => $site_host,
+      'site_uuid' => $site_uuid,
+    ];
+
+    // Additional information about the relaxed endpoint.
+    $relaxed_config = \Drupal::config('relaxed.settings');
+    $relaxed_root = $relaxed_config->get('api_root');
+    $relaxed_password = \Drupal::service('relaxed.sensitive_data.transformer')->get($relaxed_config->get('password'));
+
+    // We create an encoded uri for this site.
+    $uri = new Uri($site_host . $relaxed_root);
+    $uri = $uri->withUserInfo(
+      $relaxed_config->get('username'),
+      $relaxed_password
+    );
+    $encoded = \Drupal::service('relaxed.sensitive_data.transformer')->set((string) $uri);
+    $body['endpoint_uri'] = $encoded;
```

Comment:
Does it make sense to send the uri encoded? Data must be transmitted https encoded anyway. That might be problematic if the encoding usees the hash-salt which differs per site?

But let's ensure the remote endpoint uses https and issue some warning if it does not. Let's do not fail so we can have local testing still.